File: Leetcode_POTD_2025/8_August/3195_Find_the_Minimum_Area_to_Cover_All_Ones_I.java
class Solution {
    public int minimumArea(int[][] grid) {
        int rows = grid.length;
        int cols = grid[0].length;
        int minR = rows, maxR = -1;
        int minC = cols, maxC = -1;

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                if (grid[i][j] == 1) {
                    minR = Math.min(minR, i);
                    maxR = Math.max(maxR, i);
                    minC = Math.min(minC, j);
                    maxC = Math.max(maxC, j);
                }
            }
        }
        
        return (maxR - minR + 1) * (maxC - minC + 1);
    }
}

File: Leetcode_POTD_2025/8_August/3195_Find_the_Minimum_Area_to_Cover_All_Ones_I.py
class Solution(object):
    def minimumArea(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        rows = len(grid)
        cols = len(grid[0])
        min_r, max_r = rows, -1
        min_c, max_c = cols, -1

        for r in range(rows):
            for c in range(cols):
                if grid[r][c] == 1:
                    min_r = min(min_r, r)
                    max_r = max(max_r, r)
                    min_c = min(min_c, c)
                    max_c = max(max_c, c)
        
        return (max_r - min_r + 1) * (max_c - min_c + 1)
